modules/wallet/seed.go

```diff
@@ -287,45 +287,98 @@ func (w *Wallet) SweepSeed(seed modules.Seed) (payout types.Currency, err error)
-		// construct an input for the output
+		// construct a siacoin input that spends the output
 		sk := generateSpendableKey(seed, output.seedIndex)
-		txn.SiacoinInputs = append(txn.SiacoinInputs, types.SiacoinInput{
+		tb.AddSiacoinInput(types.SiacoinInput{
 			ParentID:         types.SiacoinOutputID(output.id),
-			UnlockConditions: sk.UnlockConditions, // TODO: check timelock, etc.
+			UnlockConditions: sk.UnlockConditions,
 		})
 		// add a signature for the input
-		swept = swept.Add(output.value)
+		sweptCoins = sweptCoins.Add(output.value)
+	}
+	for _, output := range s.siafundOutputs {
+		// construct a siafund input that spends the output
+		sk := generateSpendableKey(seed, output.seedIndex)
+		tb.AddSiafundInput(types.SiafundInput{
+			ParentID:         types.SiafundOutputID(output.id),
+			UnlockConditions: sk.UnlockConditions,
+		})
+		// add a signature for the input
+		sweptFunds = sweptFunds.Add(output.value)
 	}
 
-	// estimate the transaction size. NOTE: this equation doesn't account for
-	// other fields in the transaction, but since we are multiplying by
-	// maxFee, lowballing is ok
+	// estimate the transaction size and fee. NOTE: this equation doesn't
+	// account for other fields in the transaction, but since we are
+	// multiplying by maxFee, lowballing is ok
 	estTxnSize := len(s.siacoinOutputs) * outputSize
 	estFee := maxFee.Mul64(uint64(estTxnSize))
-	if estFee.Cmp(swept) >= 0 {
-		return types.Currency{}, errors.New("transaction fee exceeds value of swept outputs")
+	tb.AddMinerFee(estFee)
+
+	// calculate total siacoin payout
+	if sweptCoins.Cmp(estFee) > 0 {
+		coins = sweptCoins.Sub(estFee)
 	}
-	txn.MinerFees = append(txn.MinerFees, estFee)
+	funds = sweptFunds
+
+	switch {
+	case coins.IsZero() && funds.IsZero():
+		// if we aren't sweeping any coins or funds, then just return an
+		// error; no reason to proceed
+		return types.Currency{}, types.Currency{}, errors.New("transaction fee exceeds value of swept outputs")
+
+	case !coins.IsZero() && funds.IsZero():
+		// if we're sweeping coins but not funds, add a siacoin output for
+		// them
+		tb.AddSiacoinOutput(types.SiacoinOutput{
+			Value:      coins,
+			UnlockHash: uc.UnlockHash(),
+		})
 
-	// add the recipient output, equal to the sum of the inputs minus the
-	// transaction fee
-	payout = swept.Sub(estFee)
-	txn.SiacoinOutputs = append(txn.SiacoinOutputs, types.SiacoinOutput{
-		Value:      payout,
-		UnlockHash: uc.UnlockHash(),
-	})
+	case coins.IsZero() && !funds.IsZero():
+		// if we're sweeping funds but not coins, add a siafund output for
+		// them. This is tricky because we still need to pay for the
+		// transaction fee, but we can't simply subtract the fee from the
+		// output value like we can with swept coins. Instead, we need to fund
+		// the fee using the existing wallet balance.
+		tb.AddSiafundOutput(types.SiafundOutput{
+			Value:      funds,
+			UnlockHash: uc.UnlockHash(),
+		})
+		err = tb.FundSiacoins(estFee)
```

Comment:
Probably better to tell the user that they need more SC, you don't want SF stuck in limbo